pkg/oidc/login_handler.go

```diff
@@ -0,0 +1,166 @@
+<h1>Login Successful</h1>
+<p>You have successfully logged in to Skpr.</p>
+<p>Please return to the Skpr console.</p>
+<pre><code>{{.}}</code></pre>
+</div></div>
+</body>
+</html>
+`
+)
+
+// LoginHandler struct
+type LoginHandler struct {
+	cognitoConfig       config.Config
+	oauth2Config        oauth2.Config
+	tokensCache         oauth.TokenCache
+	credentialsResolver awscreds.CredentialsResolver
+}
+
+// NewLoginHandler creates a new login handler
+func NewLoginHandler(config *config.Config, tokensCache oauth.TokenCache, credentialsResolver *awscreds.CredentialsResolver) *LoginHandler {
+	endpoint := oauth2.Endpoint{
+		AuthStyle:oauth2.AuthStyleInParams,
+		AuthURL: config.AuthURL,
+		TokenURL: config.TokenURL,
+	}
+	return &LoginHandler{
+		cognitoConfig: *config,
+		oauth2Config: oauth2.Config{
+			RedirectURL:  redirectURL,
+			ClientID:     config.ClientID,
+			ClientSecret: config.ClientSecret,
+			Scopes:       []string{scopes},
+			Endpoint:     endpoint,
+		},
+		tokensCache:         tokensCache,
+		credentialsResolver: *credentialsResolver,
+	}
+}
+
+// GetAuthCodeURL gets the authorisation code URL.
+func (l *LoginHandler) GetAuthCodeURL() (string, string) {
+	state := rand.String(8)
+	return l.oauth2Config.AuthCodeURL(state, oauth2.AccessTypeOffline), state
+}
+
+// Handle handles the OAuth2 code flow.
+func (l *LoginHandler) Handle(state string) (awscreds.Credentials, error) {
+	code, respState, err := l.getCode(":8080")
```

Comment:
It would be nice to detect a free port and use that - https://github.com/phayes/freeport

Reply:
Unfortunately, you need to pre-register the callback urls (including ports).

Reply:
I've made the port configurable. Hope this is enough?